### ISSUE TITLE
spidermonkify.py can't find *.wast.mappedGlobals

### ISSUE THREAD

**Issue by floooh** [WebAssembly/binaryen#713]
Hi Alon, I currently get an error when building the WASM version of my Oryol samples with the native-wasm method. It looks like spidermonkify.py expects a mappedGlobals file, which isn't generated anymore(?). It might already be that way for a while, since I'm only building WASM from time to time.

Here's the error output:

```
Traceback (most recent call last):
  File "/Users/floh/projects/binaryen/scripts/spidermonkify.py", line 52, in <module>
    shutil.copyfile(wast_target + '.mappedGlobals', wasm_target + '.mappedGlobals')
  File "/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/shutil.py", line 82, in copyfile
    with open(src, 'rb') as fsrc:
IOError: [Errno 2] No such file or directory: '/Users/floh/projects/fips-deploy/oryol/wasm-ninja-release/CoreHello.wast.mappedGlobals'
Traceback (most recent call last):
  File "/Users/floh/projects/fips-sdks/osx/emsdk_portable/emscripten/incoming/em++", line 16, in <module>
    emcc.run()
  File "/Users/floh/projects/fips-sdks/osx/emsdk_portable/emscripten/incoming/emcc.py", line 1982, in run
    subprocess.check_call([shared.PYTHON, os.path.join(binaryen_scripts, script), js_target, wasm_text_target], env=script_env)
  File "/usr/local/Cellar/python/2.7.12/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 541, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['/usr/local/opt/python/bin/python2.7', u'/Users/floh/projects/binaryen/scripts/spidermonkify.py', '/Users/floh/projects/fips-deploy/oryol/wasm-ninja-release/CoreHello.js', '/Users/floh/projects/fips-deploy/oryol/wasm-ninja-release/CoreHello.wast']' returned non-zero exit status 1
ninja: build stopped: subcommand failed.
```

And here's the emscripten linker command line generating that output:
```
: && /Users/floh/projects/fips-sdks/osx/emsdk_portable/emscripten/incoming/em++   --em-config /Users/floh/projects/fips-sdks/osx/emsdk_portable/.emscripten --cache /Users/floh/projects/fips-sdks/osx/emsdk_portable/.emscripten_cache  -fno-exceptions -fno-rtti -std=c++11 -stdlib=libc++ -fstrict-aliasing -Wall -Wno-multichar -Wextra -Wno-unused-parameter -Wno-unknown-pragmas -Wno-ignored-qualifiers -Wno-long-long -Wno-overloaded-virtual -Wno-deprecated-writable-strings -Wno-unused-volatile-lvalue -Wno-inconsistent-missing-override -Wno-warn-absolute-paths -Wno-expansion-to-defined -O3 -DNDEBUG  --em-config /Users/floh/projects/fips-sdks/osx/emsdk_portable/.emscripten --cache /Users/floh/projects/fips-sdks/osx/emsdk_portable/.emscripten_cache  --memory-init-file 1 -s TOTAL_MEMORY=134217728 -s ERROR_ON_UNDEFINED_SYMBOLS=1 -s EXPORTED_FUNCTIONS="['_main','_enter_fullscreen','_enter_soft_fullscreen']" -s NO_EXIT_RUNTIME=1 -s NO_FILESYSTEM=1 -s DISABLE_EXCEPTION_CATCHING=1 -s BINARYEN=1 -s 'BINARYEN_METHOD="native-wasm"' -s 'BINARYEN_SCRIPTS="spidermonkify.py"' -s GLOBAL_BASE=1000 -s ALIASING_FUNCTION_POINTERS=0  -O3  --llvm-lto 1 -s OUTLINING_LIMIT=60000 --closure 0 -s ASSERTIONS=0 code/Samples/CoreHello/CMakeFiles/CoreHello.dir/CoreHello.cc.o  -o /Users/floh/projects/fips-deploy/oryol/wasm-ninja-release/CoreHello.html  code/Modules/Core/libCore.a && :
```

Is this something wrong on my side? It's not high-prio for me right now, I want to setup a new Oryol 'bleeding edge' sample webpage which has all samples with WebGL2 render backend in asm.js and WASM. For now I can simply upload an asm.js only version and add the wasm samples later.

Thanks :)
-Floh

**Comment by floooh** [comment]
PS: the output directory only has the following files:

```
➜  oryol (webgl2) ✗ ls /Users/floh/projects/fips-deploy/oryol/wasm-ninja-release/
CoreHello.asm.js CoreHello.js     CoreHello.wast
```

**Comment by kripken** [comment]
Yeah, things have changed and that mappedGlobals file is obsolete. spidermonkify is out of date in other words. It is deprecated at this point, in fact, and I removed it from the docs - unless you found somewhere it is still mentioned?

The current status is that browsers just got updated to 0xc. But there isn't currently a way to build code to 0xc, and spidermonkify can't help (hacks like spidermonkify only work if spidermonkey can read a text file binaryen emits, but the 0xc changes are just far too big for that workaround). However, binaryen passes the 0xc spec tests so we should be close to being able to emit code that runs in browsers. @dschuff is working on the last binary format pieces.
